words/scanner_test.go

```diff
@@ -381,3 +380,33 @@ func BenchmarkUnicodeSegments(b *testing.B) {
 		b.ReportMetric(float64(c), "tokens")
 	}
 }
+
+func TestScannerMaxTokenLen(t *testing.T) {
+	// Build a string longer than the max token length
+	str := ""
+	for i := 0; i < words.MaxTokenLengthDefault + 5; i++ {
+		if i == words.MaxTokenLengthDefault {
+			str += "."
```

Comment:
Looks like this test checks that we cleanly tokenize if the "." comes right after the default length, but what is the behavior if it's after that? Like if we have (a*255)aaa.aaa (where there are more initial a's than max default). Maybe add a test to show that behavior.

Reply:
Sg, just updated and added another test